extensions/data-transfer/portability-data-transfer-solid/src/main/java/org/datatransferproject/transfer/solid/contacts/SolidContactsImport.java

```diff
@@ -0,0 +1,318 @@
+package org.datatransferproject.transfer.solid.contacts;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import ezvcard.Ezvcard;
+import ezvcard.VCard;
+import ezvcard.parameter.EmailType;
+import ezvcard.parameter.TelephoneType;
+import ezvcard.property.Email;
+import ezvcard.property.Note;
+import ezvcard.property.Organization;
+import ezvcard.property.Photo;
+import ezvcard.property.StructuredName;
+import ezvcard.property.Telephone;
+import ezvcard.property.Url;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.vocabulary.DCTerms;
+import org.apache.jena.vocabulary.DC_11;
+import org.apache.jena.vocabulary.RDF;
+import org.apache.jena.vocabulary.VCARD4;
+import org.datatransferproject.spi.transfer.provider.ImportResult;
+import org.datatransferproject.spi.transfer.provider.Importer;
+import org.datatransferproject.transfer.solid.SolidUtilities;
+import org.datatransferproject.types.common.models.contacts.ContactsModelWrapper;
+import org.datatransferproject.types.transfer.auth.CookiesAndUrlAuthData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SolidContactsImport implements Importer<CookiesAndUrlAuthData, ContactsModelWrapper> {
+  //See https://www.w3.org/TR/vcard-rdf/ for details.
+
+  private static final Logger logger = LoggerFactory.getLogger(SolidContactsExport.class);
+  private static final String TEST_SLUG_NAME = "ImportedAddressBook";
+  private static final String BASE_DIRECTORY = "/inbox/";
+  static final String IMPORTED_ADDRESS_BOOK_PATH = BASE_DIRECTORY + TEST_SLUG_NAME + "/";
```

Comment:
move below the other private static members. Add annotation if it's package private for testing.